editor/src/core/shared/template-path.ts

```diff
@@ -241,18 +238,28 @@ export function scenePathForPath(path: TemplatePath): ScenePath {
   }
 }
 
-export function elementPathForPath(path: ScenePath): StaticElementPath
-export function elementPathForPath(path: StaticInstancePath): StaticElementPath
-export function elementPathForPath(path: InstancePath): ElementPath
-export function elementPathForPath(path: TemplatePath): ElementPath
-export function elementPathForPath(path: TemplatePath): ElementPath {
-  if (isScenePath(path)) {
-    return path.sceneElementPath
+export function instancePathForScenePath(path: ScenePath): StaticInstancePath {
+  const lastElementPath = last(path.sceneElementPaths)
+  if (lastElementPath == null) {
+    return emptyInstancePath
   } else {
-    return path.element
+    const targetSceneElementPaths = dropLast(path.sceneElementPaths)
+    const targetScenePath = scenePath(targetSceneElementPaths)
+    return staticInstancePath(targetScenePath, lastElementPath)
   }
 }
 
+export function scenePathFromInstancePath(path: InstancePath): ScenePath {
```

Comment:
This function is the inverse of `instancePathForScenePath`, but we also have `scenePathForPath` which has completely different functionality, so the naming is becoming a bit confusing. Right now the use of the 3 different functions are as follows (just pretend equality of objects is a thing :grinning:):

```
const a = { 
  type: 'scenepath',
  sceneElementPaths: [
    ['storyboard', 'scene'],
    ['app', 'card'],
  ]
}

const b = {
  type: 'instancepath',
  scene: {
    type: 'scenepath',
    sceneElementPaths: [
      ['storyboard', 'scene']
    ]
  },
  element: ['app', 'card']
}

const c = { 
  type: 'scenepath',
  sceneElementPaths: [
    ['storyboard', 'scene'],
  ]
}

instancePathForScenePath(a) == b // true
scenePathFromInstancePath(b) == c // true
scenePathForPath(b) == c // true
```

Reply:
I think it's worth adding some comments to these functions, to help us navigate the code in the future

Reply:
uhmm actually, shouldn't `scenePathFromInstancePath(b) == c` be instead `scenePathFromInstancePath(b) == a`?

Reply:
Yeah agreed. I was hoping someone would pitch me some much clearer names :grinning: 
I'll also add tests for each of the functions to help aid in the documentation of them.

Reply:
> uhmm actually, shouldn't `scenePathFromInstancePath(b) == c` be instead `scenePathFromInstancePath(b) == a`?

Dunno what you're talking about, that's exactly what it says

Reply:
Ok I have renamed the 3 functions to add clarity, inserted a comment at the start of them each, and added tests for them in a single `describe` block that should also help demonstrate their differences.